utils.go

```diff
@@ -1,45 +1,21 @@
 package mnemosyne
 
 import (
-	"bytes"
-	"compress/zlib"
-	"io"
+	"fmt"
 	"strings"
 )
 
-func MakeKey(keys ...string) string {
-	return strings.Join(keys, ";")
+type AmnesiaError struct {
```

Comment:
@RRostami `AmnesiaError` should not be exposed outside. consider changing it to local error and return `ErrCacheMiss` instead.

Reply:
Done.